pkg/planner/core/logical_datasource.go

```diff
@@ -126,6 +127,74 @@ func (ds DataSource) Init(ctx base.PlanContext, offset int) *DataSource {
 	return &ds
 }
 
+// ************************ start implementation of HashEquals interface ************************
+
+// Hash64 implements base.HashEquals interface.
+func (ds *DataSource) Hash64(h base2.Hasher) {
```

Comment:
@winoros @elsa0520 
current logical operator only care about some key elements to do the hash64

for example: astHints are not considered here, the start point is what the minimum we need to determinate this ds is unique one.

Reply:
not sure for other elements, agile for adjustment in future